### DISCUSSION TITLE
Continuous stream of data

### DISCUSSION THREAD

**Discussion by alexanderfitu** [jblance/mpp-solar#91] [General]
As a follow up for the issue I raised.

You mention the best way would be to have a python script rather than the complex bash script I have written.

You mention the script needs to loop and maintain state, at the moment it runs one iterance every 30 seconds, and I have tried to trap every failure mode with an ending statement the kills the relay and terminates the script. 

Cron calls the script every 30 seconds.

I suppose if I were to recreate in python, I could use the snippets of your code to make a connection to the bms units and only obtain the data I need. 

Just need to figure out how to make a continous connection like the app does

**Comment by jblance** [comment]
Yeah, well the final state would be reasonably complicated. I'd go for a daemon or service as the target
If you are rewriting in python I'd recommend starting by reproducing your current bash script in python (and just calling mppsolar as an external executable) 
Then get that running as a service
Then we could look at ripping the relevant bits out of mpp-solar and putting them in your python code directly and allowing for a data stream 
Fortunately the jkbms code uses a callback so it is actually well suited to this
